src/function/abs.rs

```diff
@@ -0,0 +1,73 @@
+use super::*;
+use crate::array::ArrayImpl;
+use crate::function::FunctionError::{InvalidDataTypes, InvalidParameters};
+
+pub struct AbsFunction {}
+
+impl Function for AbsFunction {
+    fn name(&self) -> &str {
+        "abs"
+    }
+
+    fn return_types(
+        &self,
+        input_types: &[PhysicalDataTypeKind],
+    ) -> Result<Vec<PhysicalDataTypeKind>, FunctionError> {
+        if input_types.len() != 1 {
+            return Err(InvalidDataTypes("The column size should be 1".to_string()));
+        }
+        match &input_types[0] {
+            PhysicalDataTypeKind::Int32 => Ok(vec![PhysicalDataTypeKind::Int32]),
+            PhysicalDataTypeKind::Int64 => Ok(vec![PhysicalDataTypeKind::Int64]),
+            PhysicalDataTypeKind::Float64 => Ok(vec![PhysicalDataTypeKind::Float64]),
+            _ => Err(InvalidDataTypes("Data type is not supported".to_string())),
+        }
+    }
+
+    fn execute(&self, input: &DataChunk) -> Result<DataChunk, FunctionError> {
+        if input.column_count() != 1 {
+            return Err(InvalidParameters("The column size should be 1".to_string()));
+        }
+
+        let arr = input.array_at(0);
+        match &arr {
+            ArrayImpl::Int32(i32_arr) => {
```

Comment:
I'd prefer having a framework to vectorize and dispatch by type. Need more consideration.

Reply:
Yes, we should consider it carefully. I will not implement more functions before our discussion.